File: 프로그래머스/lv2/권채린/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
// 12233455

// 서로 다른 종류를 최소화 => 최빈값 체크 
using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    int m = *max_element(tangerine.begin(), tangerine.end());
    vector<int> size(m+1,0);

    for(int i=0; i<tangerine.size(); i++) {
        size[tangerine[i]]++;
    }
    
    sort(size.begin(), size.end(), greater<int>());
    
    int idx=0;
    while(k>=1) {
        k -=size[idx];
        answer++;
        idx++;
    }
    
    return answer;
}

File: 프로그래머스/lv2/이재표/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer,Integer>type=new HashMap<>();
        for(int i=0;i<tangerine.length;i++){
            type.put(tangerine[i],type.getOrDefault(tangerine[i],0)+1);
        }
        Set<Integer> keys = type.keySet();
        List<Integer> keyList = new ArrayList<>(keys);
        keyList.sort((o1, o2) -> type.get(o2).compareTo(type.get(o1)));
        int answer = 0;
        for(Integer key:keyList){
            if(k<=0){
                break;
            }
            k-=type.get(key);
            answer++;
        }
        return answer;
    }
}
